### DISCUSSION TITLE
Visualizing diagnostic key (string that is "semantically" a float) in a plot.

### DISCUSSION THREAD

**Discussion by AlboAlby00** [foxglove/community#953] [Q&A] [answered]
I'm currently publishing how much CPU usage is being used by a certain process in the diagnostics. The message diagnostic_msgs/KeyValue.msg has two fields string key and string value. The value is a string due to the message definition, but it is "semantically" a float. I'd like to visualize these values in a plot. Is there any way of doing it directly with the diagnostic msg string, without publishing a custom message with the same information published as a float?

**Comment by jtbandes** [comment, marked as answer]
Yes, you can click the 📈 icon next to the value, which will open a Plot panel to plot it:

<img width="401" alt="image" src="https://github.com/foxglove/community/assets/14237/255b81c8-6b95-4826-84e9-5d716430c17e">

<img width="780" alt="image" src="https://github.com/foxglove/community/assets/14237/f74a8095-4248-468c-a506-80d39eaf5034">

**Reply by karl-schulz** [reply]
This does not work if the numeric values have units added to them :/

**Comment by AlboAlby00** [comment]
Thank you very much for the quick and clear answer, I'd also be interesting to be able to plot also diagnostics with the "%" symbols ( "10.0%" ).
This is not supported yet, right?

**Reply by karl-schulz** [reply]
To add on this: It would be good to be able to ignore any kind of unit suffixes. Plotjuggler does this out of the box ;)

**Reply by 2metres** [reply]
Not sure about converting strings into values here but I could envision using something like [d3-format](https://observablehq.com/@d3/d3-format)'s syntax here.

**Reply by defunctzombie** [reply]
The best approach we offer for such bespoke transformations of strings like removing `%` or suffix is via `user-scripts`. You can transform the message data to strip away whatever parts you don't need or transform the values.

**Reply by karl-schulz** [reply]
Thanks for the responses.
